File: RUBY/Cap07/c07ex01.rb
# c07ex01.rb

#!/usr/bin/ruby

def rotsoma
  puts "\n"
  puts "Rotina de Adicao"
  puts "\n"
  print "Entre o 1o. valor: "
  a1 = gets.chomp.to_f
  print "Entre o 2o. valor: "
  b1 = gets.chomp.to_f
  puts "\n"
  r1 = a1 + b1
  printf "O resultado da operacao equivale a: %.2f\n", r1
  puts "\n"
end

def rotsubtracao
  puts "\n"
  puts "Rotina de Subtracao"
  puts "\n"
  print "Entre o 1o. valor: "
  a2 = gets.chomp.to_f
  print "Entre o 2o. valor: "
  b2 = gets.chomp.to_f
  puts "\n"
  r2 = a2 - b2
  printf "O resultado da operacao equivale a: %.2f\n", r2
  puts "\n"
end

def rotmultiplicacao
  puts "\n"
  puts "Rotina de Multiplicacao"
  puts "\n"
  print "Entre o 1o. valor: "
  a3 = gets.chomp.to_f
  print "Entre o 2o. valor: "
  b3 = gets.chomp.to_f
  puts "\n"
  r3 = a3 * b3
  printf "O resultado da operacao equivale a: %.2f\n", r3
  puts "\n"
end

def rotdivisao
  puts "\n"
  puts "Rotina de Divisao"
  puts "\n"
  print "Entre o 1o. valor: "
  a4 = gets.chomp.to_f
  print "Entre o 2o. valor: "
  b4 = gets.chomp.to_f
  puts "\n"
  if (b4 == 0)
    puts "O resultado da operacao equivale a: ERRO"
  else
    r4 = a4 / b4
    printf "O resultado da operacao equivale a: %.2f\n", r4
  end
  puts "\n"
end

opcao = 0
while (opcao != 5)
  puts "CALCULADORA - V1"
  puts "\n"
  puts "[1] - Adicao"
  puts "[2] - Subtracao"
  puts "[3] - Multiplicacao"
  puts "[4] - Divisao"
  puts "[5] - Fim de Programa"
  puts "\n"
  print "Escolha uma opcao: "
  opcao = gets.chomp.to_i
  if (opcao != 5)
    case (opcao)
    when 1
      rotsoma
    when 2
      rotsubtracao
    when 3
      rotmultiplicacao
    when 4
      rotdivisao
    else
      puts "\n"
      puts "Opcao invalida - Tente novamente"
      puts "\n"
    end
  end
end

File: KOTLIN/Cap07/c07ex01.kt
// c07ex01.kt

fun rotAdicao() {
    println()
    println("Rotina de Adicao")
    println()
    print("Entre o 1o. valor: ")
    val a1 = readln().toFloat()
    print("Entre o 2o. valor: ")
    val b1 = readln().toFloat()
    println()
    val r1 = a1 + b1
    println("O resultado da operacao equivale a: $r1")
    println()
}

fun rotSubtracao() {
    println()
    println("Rotina de Subtracao")
    println()
    print("Entre o 1o. valor: ")
    val a2 = readln().toFloat()
    print("Entre o 2o. valor: ")
    val b2 = readln().toFloat()
    println()
    val r2 = a2 - b2
    println("O resultado da operacao equivale a: $r2")
    println()
}

fun rotMultiplicacao() {
    println()
    println("Rotina de Multiplicacao")
    println()
    print("Entre o 1o. valor: ")
    val a3 = readln().toFloat()
    print("Entre o 2o. valor: ")
    val b3 = readln().toFloat()
    println()
    val r3 = a3 * b3
    println("O resultado da operacao equivale a: $r3")
    println()
}

fun rotDivisao() {
    println()
    println("Rotina de Divisao")
    println()
    print("Entre o 1o. valor: ")
    val a4 = readln().toFloat()
    print("Entre o 2o. valor: ")
    val b4 = readln().toFloat()
    println()
    if (b4 == 0.0f)
        println("O resultado da operacao equivale a: ERRO")
    else {
        val r4 = a4 / b4
        println("O resultado da operacao equivale a: $r4")
    }
    println()
}

fun main() {
    var option = 0
    while (option != 5) {
        println("CALCULADORA - V1")
        println()
        println("[1] - Adicao")
        println("[2] - Subtracao")
        println("[3] - Multiplicacao")
        println("[4] - Divisao")
        println("[5] - Fim de Programa")
        println()
        print("Escolha uma opcao: ")
        option = readln().toInt()

        if (option != 5) {
            when (option) {
                1 -> rotAdicao()
                2 -> rotSubtracao()
                3 -> rotMultiplicacao()
                4 -> rotDivisao()
                else -> {
                    println()
                    println("Opcao invalida - Tente novamente")
                    println()
                }
            }
        }
    }
}
